Add external name/ip and load balancer ingress to service collector

Patstrom opened the pull request:
**What this PR does / why we need it**:

Add some more metrics about services. While these metrics don't change much having them collected by kube-state-metrics would allow admins to share e.g AWS ELB DNS names associated with services on platforms like Grafana.


Fixes #544

brancz commented:
Could external name and load balancer ip not be part of an info metric? Not sure a separate metric is necessary. For the arrays, I don't see another way other than separate metrics.

brancz commented on pkg/collectors/service.go:
```diff
@@ -110,5 +138,33 @@ func generateServiceMetrics(obj interface{}) []*metrics.Metric {
+			addGauge(descServiceSpecExternalIP, 1, external_ip)
+		}
+	} else {
+		addGauge(descServiceSpecExternalIP, 0, "") // Add empty for completeness
```
I don't think we should do these, if there is no external IP/loadbalancer ingress IP then there should be no metric

Patstrom commented:
I've removed the empty metrics and moved everything in `Service.Spec` to the `service_info` metric.

I've also signed the CLA but I must've done something wrong. I'll try again.

Patstrom commented:
/check-cla

I signed it

brancz commented:
/check-cla

brancz commented:
@mxinden what are your thoughts on adding features in addition to the architectural changes we have done?

Patstrom commented:
I rebased this on master to solve conflicts. What is the status on this?

brancz commented:
/lgtm

mxinden commented on pkg/collectors/service.go:
```diff
@@ -53,12 +53,40 @@ var (
 		nil,
 	)
 
+	descServiceExternalName = metrics.NewMetricFamilyDef(
```
`descServiceExternalName` is defined here, but not used anywhere. Am I missing something?

mxinden commented on pkg/collectors/service.go:
```diff
@@ -53,12 +53,40 @@ var (
+		nil,
+	)
+
+	descServiceLoadBalancerIP = metrics.NewMetricFamilyDef(
```
`descServiceLoadBalancerIP` is defined here but not used anywhere. Am I missing something?

Patstrom commented:
`external_name` and `load_balancer_ip` were both added to `kube_service_info` in 7901bb8. I must have forgotten to remove the old metric families. They should be safe to remove
